res/controllers/Numark_DJ2GO2_Touch_scripts.js

```diff
@@ -0,0 +1,192 @@
+// Declare the variable for your controller and assign it to an empty object
+var DJ2GO2Touch = {};
+
+// Mixxx calls this function on startup or when the controller
+// is enabled in the Mixxx Preferences
+DJ2GO2Touch.init = function() {
+    // create an instance of your custom Deck object for each side of your controller
+    DJ2GO2Touch.leftDeck = new DJ2GO2Touch.Deck([1], 0);
+    DJ2GO2Touch.rightDeck = new DJ2GO2Touch.Deck([2], 1);
+};
```

Comment:
```suggestion
    midi.sendSysexMsg([0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7], 0);
};
```
Not sure if your controller supports it, but if you add this, your controller should send the values of all of its controls. This way, Mixxx should be instantly synced with the controlled when the mapping was loaded.

Reply:
I just tried adding this, it didnt seem to make a difference, is there anything I should look out for in the controller debug output?

Reply:
```
Debug [Controller]: DJ2GO2 Touch MIDI 1: outgoing: status 0x95 (ch 6, opcode 0x9), ctrl 0x23, val 0x00
Debug [Controller]: DJ2GO2 Touch MIDI 1: outgoing: 7 byte sysex: [F0 00 20 7F 03 01 F7]
Debug [Controller]: DJ2GO2 Touch MIDI 1: outgoing: status 0x94 (ch 5, opcode 0x9), ctrl 0x33, val 0x7F
Debug [Controller]: DJ2GO2 Touch MIDI 1: outgoing: status 0x94 (ch 5, opcode 0x9), ctrl 0x34, val 0x7F
Debug [Controller]: DJ2GO2 Touch MIDI 1: outgoing: status 0x94 (ch 5, opcode 0x9), ctrl 0x35, val 0x7F
```

I spotted it, but not sure ...

Reply:
The controller should usually respond with a bunch of messages. We have a wiki page on this if you need more context: https://github.com/mixxxdj/mixxx/wiki/serato_sysex

Reply:
I understand now, it mostly works, the faders are accurate to their state on the controller, potentiometers aren't though, I'll commit that now.

Reply:
Mhmm, thats strange...